processor/internal/transformer/destination_transformer/embedded/warehouse/transformer.go

```diff
@@ -162,7 +144,7 @@ func (t *Transformer) Transform(_ context.Context, clientEvents []types.Transfor
 		g.Go(func() error {
 			r, err := t.processWarehouseMessage(c, &event)
 			if err != nil {
-				results <- transformerResponseFromErr(&eventMetadata, err)
+				results <- transformerResponseFromErr(&eventMetadata, embeddedutils.GetValidationErrorStatTags(event.Destination), err)
```

Comment:
While deep copying, got this bug. We need to send stats in case of failed events.